### DISCUSSION TITLE
streaming full page from server

### DISCUSSION THREAD

**Discussion by yatesco** [tbranyen/diffhtml#235] [Q&A] [answered]
Hi. Is there any reason I can't naively render _everything_ on the server and stream it to the client assuming:
 - a fast LAN with relatively 0 latency
 - a fast server with relatively 0 latency
 - websockets with relatively 0 latency

A "vaadin/Apache Wicket" lite sort of thing?

I'm thinking of an architecture where the server holds the state (minus, scrolling positions and also, possibly, input state and form validation state) and the client is the thinest of thinest client/views. Opening a new in-app dialog sends a message to the server which redraws and sends back the entire view with the dialog box opened.

Is this madness? Obviously "is it fast enough" is always and only answered correctly by "dunno - measure", but has anyone tried this?

Thanks!

**Comment by tbranyen** [comment, marked as answer]
This should be doable, although it sounds really similar the webworker implementation I tried in the past and couldn't get right. I have not tried doing a bidirectional server render, only one way server->client.

**Reply by yatesco** [reply]
thanks Tim. That's encouraging that it could be fast enough. I was looking for a "none starter - yes/no?" which you've answered, so thanks!

**Reply by aralroca** [reply]
IMO should be great to support HTML streaming for RPCs to update the DOM transferring HTML. Example rerendering from a server action.
